### DISCUSSION TITLE
SharePoint Access Requests - Graph API and PnP

### DISCUSSION THREAD

**Discussion by Deepayy** [microsoft-search/pnp-modern-search#2617] [Q&A]
I am trying to work out the scope of approving these access requests 

![image](https://user-images.githubusercontent.com/98158373/205029727-0cb35d53-2525-49c6-8d83-7c1f28a3749e.png)

Does it limit the access via the Graph API only to the data for the user logged in to SharePoint when they visit the site to or does it allow persistent access to data for all users in the tenant?

**Comment by edarroudi** [comment]
Hey @Deepayy,

Only the logged in user can access his data. But also once you approve a permission, it can be used by any application on any SP site with the logged in user. 

[Quote from MS](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/use-aadhttpclient)

_"All permissions are granted to the whole tenant and not to a specific application that has requested them. When the administrator grants a specific permission, its added to the SharePoint Online Client Extensibility Azure AD application, which is provisioned by Microsoft in every Azure AD and which is used by the SharePoint Framework in the OAuth flow to provide solutions with valid access tokens."_

**Reply by wobba** [reply]
The Graph permissions are only needed for the Graph provider, not the SharePoint provider. So feel free not to grant them access. Exception is if you use the mgt people component.
